src/App.test.js

```diff
@@ -69,27 +85,54 @@ describe('<App />', () => {
       );
     });
 
-    it('should render sign up form and call firebase correctly', async () => {
+    it('should render sign up form and call firebase correctly for voter', async () => {
       mockUnauthenticatedUser();
       customRender();
 
-      user.click(screen.getByTestId('signup-button'));
+      fillUserData();
 
-      const emailInput = screen.getByLabelText('E-mail');
-      user.type(emailInput, 'any@email.com');
+      const signUpButton = screen.getByTestId('submit-button');
```

Comment:
Sempre tente não achar o elemento pelo `testId` é sempre a última prioridade, tente fazer algo parecido como:
```suggestion
      const signUpButton = screen.getByRole('button', { name: /cadastrar/i });
      expect(signUpButton).toBeInTheDocument();
```

Reply:
Um bom artigo pra ajudar a escolher a melhor query seria seguir essa ordem aqui:
https://testing-library.com/docs/guide-which-query

Reply:
Boa, agora entendi o motivo dele, valeu!